tests/src/test/scala/com/nvidia/spark/rapids/RmmSparkRetrySuiteBase.scala

```diff
@@ -34,6 +34,7 @@ class RmmSparkRetrySuiteBase extends FunSuite with BeforeAndAfterEach {
     }
     val deviceStorage = new RapidsDeviceMemoryStore()
     val catalog = new RapidsBufferCatalog(deviceStorage)
+    RapidsBufferCatalog.setDeviceStorage(deviceStorage)
```

Comment:
A little confused why this is needed. It looks redundant. Could you at least add in some comments explaining what is happening.